### PR TITLE
Measure toolbar units bug

### PR DESCRIPTION
**Pull request by rolson**
@philium @mhdostal please review

this fixes 2 issues:
- incorrect units selection after selecting a line feature, then going into measure area mode
- incorrect measurements after changing modes

Note - this merges into @philium's branch for autolayout fixes

### CODE REVIEW

File: Toolkit/ArcGISToolkit/MeasureToolbar.swift [hunk 1]
```diff
@@ -407,27 +413,49 @@ public class MeasureToolbar: UIToolbar, AGSGeoViewTouchDelegate {
         mapView?.sketchEditor?.clearGeometry()
     }
     
+    private var linearUnits: [AGSLinearUnit] {
```
Comment (philium) [inline Toolkit/ArcGISToolkit/MeasureToolbar.swift]: Could you please make this and `areaUnits` lazy? They are the same every time, correct?

File: Toolkit/ArcGISToolkit/MeasureToolbar.swift [hunk 2]
```diff
@@ -407,27 +413,49 @@ public class MeasureToolbar: UIToolbar, AGSGeoViewTouchDelegate {
...
+        let linearUnitIDs: [AGSLinearUnitID] = [.centimeters, .feet, .inches, .kilometers, .meters, .miles, .millimeters, .nauticalMiles, .yards]
+        return linearUnitIDs
+            .compactMap { AGSLinearUnit(unitID: $0) }
```
Comment (philium) [inline Toolkit/ArcGISToolkit/MeasureToolbar.swift]:
```suggestion
            .compactMap(AGSLinearUnit.init)
```

File: Toolkit/ArcGISToolkit/MeasureToolbar.swift [hunk 3]
```diff
@@ -407,27 +413,49 @@ public class MeasureToolbar: UIToolbar, AGSGeoViewTouchDelegate {
...
+            .sorted { $0.pluralDisplayName < $1.pluralDisplayName }
+    }
+    
+    private var areaUnits: [AGSAreaUnit] {
+        let areaUnitIDs: [AGSAreaUnitID] = [.acres, .hectares, .squareCentimeters, .squareDecimeters, .squareFeet, .squareKilometers, .squareMeters, .squareMillimeters, .squareMiles, .squareYards]
+        return areaUnitIDs
+            .compactMap { AGSAreaUnit(unitID: $0) }
```
Comment (philium) [inline Toolkit/ArcGISToolkit/MeasureToolbar.swift]:
```suggestion
            .compactMap(AGSAreaUnit.init)
```

File: Toolkit/ArcGISToolkit/MeasureToolbar.swift [hunk 4]
```diff
@@ -407,27 +414,49 @@ public class MeasureToolbar: UIToolbar, AGSGeoViewTouchDelegate {
         mapView?.sketchEditor?.clearGeometry()
     }
     
+    private lazy var linearUnits: [AGSLinearUnit] = {
+        let linearUnitIDs: [AGSLinearUnitID] = [.centimeters, .feet, .inches, .kilometers, .meters, .miles, .millimeters, .nauticalMiles, .yards]
+        return linearUnitIDs
+            .compactMap (AGSLinearUnit.init)
```
Comment (philium) [inline Toolkit/ArcGISToolkit/MeasureToolbar.swift]: The extra space is causing a linter warning.

image

```suggestion
            .compactMap(AGSLinearUnit.init)
```

File: Toolkit/ArcGISToolkit/MeasureToolbar.swift [hunk 5]
```diff
@@ -407,27 +414,49 @@ public class MeasureToolbar: UIToolbar, AGSGeoViewTouchDelegate {
...
+            .sorted { $0.pluralDisplayName < $1.pluralDisplayName }
+    }()
+    
+    private lazy var areaUnits: [AGSAreaUnit] = {
+        let areaUnitIDs: [AGSAreaUnitID] = [.acres, .hectares, .squareCentimeters, .squareDecimeters, .squareFeet, .squareKilometers, .squareMeters, .squareMillimeters, .squareMiles, .squareYards]
+        return areaUnitIDs
+            .compactMap (AGSAreaUnit.init)
```
Comment (philium) [inline Toolkit/ArcGISToolkit/MeasureToolbar.swift]:
```suggestion
            .compactMap(AGSAreaUnit.init)
```

**Comment by rolson** [comment]
I'm not sure why the linter isn't showing me warnings for those

**Pull request merged**